src/Roslyn.Suppressions/Roslyn.Suppressions/SuppressionExecutionContext.cs

```diff
@@ -55,5 +55,7 @@ private void GetProjectKey(AnalyzerOptions analyzerOptions)
         public bool IsInConnectedMode => SettingsKey != null;
 
         public string SettingsKey { get; private set; } = null;
+
+        public string Mode => SettingsKey != null ? "Connected" : "Standalone";
```

Comment:
Even it adds more complexity, I think having the mode name logic here makes the code more readable

Reply:
Then just to make it readable maybe reuse `IsInConnectedMode` instead of checking the settings key again.